app/src/main/java/com/islamey/hamza/wazaker/ui/AzkarListFragment/AzkarListAdapter.kt

```diff
@@ -0,0 +1,59 @@
+package com.islamey.hamza.wazaker.ui.AzkarListFragment
+
+
+import android.annotation.SuppressLint
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.AsyncListDiffer
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.RecyclerView
+import com.islamey.hamza.wazaker.domain.Models.AzkarListModel
+import com.islamey.hamza.wazaker.ui.OnItemListener
+import com.islamey.wazkar.databinding.SimpleCellBinding
+
+class AzkarListAdapter constructor(
+    private val onItemListener: OnItemListener
+) : RecyclerView.Adapter<AzkarListAdapter.ViewHolder>() {
+
+    inner class ViewHolder(val binding: SimpleCellBinding) :
+        RecyclerView.ViewHolder(binding.root)
+
+    private val diffCallback = object : DiffUtil.ItemCallback<AzkarListModel>() {
+        override fun areItemsTheSame(oldItem: AzkarListModel, newItem: AzkarListModel): Boolean {
+            return oldItem.name == newItem.name
+        }
+
+        override fun areContentsTheSame(oldItem: AzkarListModel, newItem: AzkarListModel): Boolean {
+            return oldItem == newItem
+        }
+    }
+
+    var differ = AsyncListDiffer(this, diffCallback)
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AzkarListAdapter.ViewHolder {
+        val view =
+            SimpleCellBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+        return ViewHolder(view)
+    }
+
+    @SuppressLint("SetTextI18n")
+    override fun onBindViewHolder(holder: AzkarListAdapter.ViewHolder, position: Int) {
+
+        val currentItem = differ.currentList[position]
+
+        holder.binding.tvTitle.text = currentItem.name
+
+
+        holder.itemView.setOnClickListener {
+            onItemListener.onItemCLicked(position)
+        }
```

Comment:
try to figure out another way to set listener instead of having it inside on bind.